### ISSUE TITLE
Title screen sample sounds like a Vinesauce corruption stream on emulator

### ISSUE THREAD

**Issue by kmeisthax** [telefang/telefang#33]
The sample code in the sound engine is a blocking routine that bitbangs samples into the volume register of two voices. This works on real hardware but fails on every emulator I've tried - including the official Nintendo ones. There are other ways to generate samples on Game Boy; we should try those.

If the issue isn't fixable, we should at least consider writing a patch file for people using our patch with the official Nintendo GBC emulator, which supports sample replacement.

This affects both original ROMs and our patch.

**Comment by Blaziken257** [comment]
This seems to be fixed in recent versions of several emulators (BGB, mGBA, Gambatte through Retroarch, and even VBA-M). This is the case for both the Japanese title screen and the English one ("Let's Telefang!") However, it is still an issue with 3DS Virtual Console.

**Comment by Blaziken257** [comment]
Surprise, the Analogue Pocket has issues with Telefang's title screen (as of firmware v1.1 beta 7). Most of the time, the sample is silent, but there's about a 25% or so chance that it plays correctly whenever the title screen appears.

**Comment by Blaziken257** [comment]
mGBA broke something in 0.10.x and voice samples are once again a problem in this emulator. This _also_ applies to Medarot 3, 4, and 5, which were also developed by Natsume.

**Comment by Blaziken257** [comment]
Just recently, the user nitro2k01 dropped a bombshell (albeit for Medarot 3) with an explanation [here](https://github.com/Medabots/medarot3/issues/183#issuecomment-2016586372)... and it turns out that the code for voice samples is actually _buggy_. Apparently the fact that it works at all is "dumb luck" according to him, with a very thorough explanation as to why this is the case. He offered to rewrite the sample playback code from scratch for M3 if there's demand for it.

While the code in Telefang isn't 100% identical, it is still similar enough (see [here](https://github.com/Medabots/medarot3/blob/tr_EN/game/src/sound/samples.asm)). If a fix for Medarot 3 can be made, it might be possible to port this over to Telefang so that it doesn't have issues with less accurate emulators!

**Comment by Maverynthia** [comment]
Anyway to get a version that DOESN'T change the title screen voice sample. Might be better sounding (even if it's jank)

**Comment by Maverynthia** [comment]
I just did the Japanese one sounds fine. (BGB64) Still, making the sound patch separate is a good idea.

**Comment by andwhyisit** [comment]
"Let's Telefang"

**Comment by Maverynthia** [comment]
Yeah can't tell that, the English sounds wrong compared to the Japanese "Let's Telefang". Still. Having one without the dub would be nice.
